lib/src/global_packages.dart

```diff
@@ -750,7 +750,7 @@ To recompile executables, first run `global deactivate ${dep.name}`.
         assert(p.isAbsolute(snapshot));
         invocation = '''
 if exist "$snapshot" (
-  dart "$snapshot" %*
+  call dart "$snapshot" %*
```

Comment:
If `dart` from `PATH` is actually `dart.bat` then using [`call`](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/call) makes a difference, because invoking a bat script from another bat script will stop the original bat script from working.

How that triggers this to be run twice I don't understand.

hmm, can this cause problems when `dart` is not a `dart.bat` file, which is the case when installed from the Dart SDK.

Reply:
I have no clue how but from several hours of investigating, I found that even when the if branch of the `errorlevel 253` if statement runs, the `goto error` line doesn't run. I put an echo before and after and both are printed -- it's almost like the goto is skipped. Using `call` somehow prevents this from happening. My only guess is that something in `dart.bat`, or one of the batch files it calls, must be changing something or setting a conflicting label, and that using `call` tells CMD to use Dart in a separate context. I couldn't find anything concrete though.

When `dart` is really `dart.exe` (in `flutter/bin/cache/dart-sdk/bin/dart.exe`, the place `dart.bat` forwards to, `call` still works. I ran all the binstub tests for both versions of PATH, making sure to close cmd.exe.

Reply:
To clarify, it isn't the `dart snapshot` that runs twice. Instead, the `goto error` line is skipped, even when the if statement is run. That causes the first `dart pub global run` command to run as well, effectively running the executable twice. 

That's why so many fixes out there (and my initial draft for this PR) simply deleted the first `dart pub global run` line. But that broke some tests, this is the proper solution.